### ISSUE TITLE
add support for writing RGB data

### ISSUE THREAD

**Issue by mattersoflight** [czbiohub/iohub#74]
We often need to export ND color overlays. @edyoshikun and I ran into this when trying to analyze zebrafish data.
I experimented and found the following:
* RGB data can be written like any other channel (https://github.com/ome/napari-ome-zarr/issues/57).
* `napari-ome-zarr` plugin requires appropriate channel metadata to render the RGB overlays. 

Example FOVs: 
`/Volumes/comp_micro/projects/automation/dataFormats/ome_zarr_demo/test_astro_RGB`

`/hpc/projects/comp_micro/projects/zebrafish-infection/2023_02_02_hummingbird_casper_GFPmacs/hsv_pos7.zarr/positions/0/0/` 

metadata needed in `.zattrs`
```
 "channels": [
            {
                "active": true,
                "color": "FF0000",
                "label": "Red",
                "window": {
                    "end": 0.5,
                    "start": 0.0
                }
            },
            {
                "active": true,
                "color": "00FF00",
                "label": "Green",
                "window": {
                    "end": 0.5,
                    "start": 0.0
                }
            },
            {
                "active": true,
                "color": "0000FF",
                "label": "Blue",
                "window": {
                    "end": 0.5,
                    "start": 0.0
                }
            }
        ],
```

Image appears as follows with napari-ome-zarr
![image](https://user-images.githubusercontent.com/2934183/222041967-58f381ba-91a9-4b98-925a-75a641a4dece.png)

**Comment by mattersoflight** [comment]
@talonchandler mentioned that napari image layer supports the display of RGB or RGBA images if the last dimension of the array has size 3 or 4. 

I see two paths forward:
option 1: `iohub` has the option to add 6D arrays for overlays. These would not conflict with NGFF spec, but allow the custom viewers we write to display.
option 2: We stick with NGFF spec of 5D arrays, but encode each RGB image into 3 channels with names such as `<overlay>_R/G/B`. These layers can be assigned the appropriate color maps and linked.

I like option-2 because it seems that we can use `napari-ome-zarr` out of the box to overlay the color images with other data.

@edyoshikun, @talonchandler , @ziw-liu, others your thoughts?

**Comment by ziw-liu** [comment]
Adding to @edyoshikun's point, I think we should implement NGFF spec to the extent possible in iohub. The custom viewers can also easily read the same data and create one RGB layer when there is a consistent channel naming scheme.

**Comment by talonchandler** [comment]
I think there are two separate items here:

**Writing RGB data** - I'm in favor of sticking to the spec. 
**Viewing RGB data** - If we're implementing napari readers then we have flexibility in how we load RGB images into layers. Staying with the spec enables napari and other reader to get RGB as layers "out of the box", and we have the flexibility to have our napari readers use the layer names to view RGB data in a single layer if desired. 

So I'm leaning towards a mix of 1 & 2: option 1's viewing so that an RGB appears as a single napari layer (via `recorder view` and/or others?), and option 2's writing part so that we adhere to the spec. 

Worth noting: napari has a menu option to split an RGB image into individual layers, but (as far as I know) no easy way to combine layers to RGB. 

<img width="260" alt="Screenshot 2023-03-29 at 10 23 43 PM" src="https://user-images.githubusercontent.com/9554101/228738736-992ee476-fe82-4124-981d-e1d0a7111729.png">

**Comment by ziw-liu** [comment]
Can this be closed by #101?

**Comment by edyoshikun** [comment]
For applications that require RGB to change the layer specs, this works:

```python 
v = napari.Viewer()
channels_names = ['Red', 'Green', 'Blue']
RBG_layers = [layer.name for layer in v.layers if any(s in layer.name for s in channels_names)]
v.layers.link_layers(RBG_layers)
```
